quandhz/resaga
# Improve setValue

Branch https://github.com/quandhz/resaga/tree/improve_setValue
Resolve #15 

### Improve setValue
```
setValue('book', 'Harry Potter'); // withValue
setValue('sold', (currentValue) => currentValue + 1); // withFunction
setValue({
  book: 'Harry Potter',
  sold: (currentValue) => currentValue + 1,
}); // withObject
```
### Callback supported
An object with the updated key/value.
If you use object and there are more than 1 key/value pair to be updated, only 1 callback will be called, when all key/value pairs have been set.
```
// Say current sold is 110
setValue({
  book: 'Harry Potter',
  sold: (currentValue) => currentValue + 1,
}, console.log); // log { book: 'Harry Potter', sold: 111 }
```

### How to test
- `npm run test:jest`
- `npm run examples:counter` and check log to see callbacks
